src/app/modules/shared-components/components/watch-button/watch-button.component.ts

```diff
@@ -1,55 +1,47 @@
-import { Component, Input, OnChanges, OnDestroy, ViewChild } from '@angular/core';
+import { Component, ViewChild } from '@angular/core';
 import { OverlayPanel } from 'primeng/overlaypanel';
-import { GroupWatchingService, StartWatchGroupInfo } from '../../../../core/services/group-watching.service';
-import { combineLatest, map, ReplaySubject } from 'rxjs';
-import { GroupRoute } from '../../../../shared/routing/group-route';
+import { GroupWatchingService } from '../../../../core/services/group-watching.service';
+import { combineLatest, map } from 'rxjs';
+import { CurrentContentService } from '../../../../shared/services/current-content.service';
+import { filter } from 'rxjs/operators';
+import { GroupInfo, isGroupInfo } from '../../../../shared/models/content/group-info';
 
 @Component({
   selector: 'alg-watch-button',
   templateUrl: './watch-button.component.html',
   styleUrls: [ './watch-button.component.scss' ],
 })
-export class WatchButtonComponent implements OnChanges, OnDestroy {
+export class WatchButtonComponent {
   @ViewChild('op') op?: OverlayPanel;
 
-  @Input() route?: GroupRoute;
-  @Input() startWatchGroupInfo?: StartWatchGroupInfo;
-
-  startWatchGroupInfo$ = new ReplaySubject<StartWatchGroupInfo>(1);
+  groupInfo$ = this.currentContentService.content$.pipe(
+    filter(isGroupInfo),
+  );
   isCurrentGroupWatched$ = combineLatest([
     this.groupWatchingService.watchedGroup$,
```

Comment:
If used on a page which could change content then when changing from a group to an item, as nothing would be emitted, the watch button related to the previous content would stay :-/
Also the two observables are not independent and so it forces you to do 2 async in cascade in the template. 

It would be cleaner to have something like.
```ts
currentGroupInfo$ = combineLatest([
  this.currentContentService.content$,
  this.groupWatchingService.watchedGroup$
]).pipe(
  map(([ currentContent, watchedGroup ) => (isGroupInfo(currentContent) ? { ...currentContent, isBeingWatched: currentContent.route.id === watchedGroup?.route.id } : undefined )
  })
);
```

Reply:
Fixed